src/contracts/BPool.sol

```diff
@@ -597,9 +601,10 @@ contract BPool is BToken, BMath, IBPool {
    * be interpreted as locked
    */
   function _setLock(bytes32 value) internal virtual {
-    assembly ("memory-safe") {
-      tstore(_MUTEX_TRANSIENT_STORAGE_SLOT, value)
-    }
+    // assembly ("memory-safe") {
+    //   tstore(_MUTEX_TRANSIENT_STORAGE_SLOT, value)
+    // }
+    _reenteringMutex = value;
   }
```

Comment:
@0xteddybear `_setLock` is virtual, so we could create a `private _reenteringMutex` (so it isn't mocked) and override this method on a higher level contract, wdyt?